control-plane/api-gateway/controllers/gateway_class_config_controller_test.go

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package controllers
+
+import (
+	"context"
+	gwv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+	"testing"
+	"time"
+
+	logrtest "github.com/go-logr/logr/testing"
+	"github.com/hashicorp/consul-k8s/control-plane/api/v1alpha1"
+	"github.com/stretchr/testify/require"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func TestGatewayClassConfigSetup(t *testing.T) {
+	require.Error(t, (&GatewayClassConfigController{}).SetupWithManager(nil))
+}
+
+func TestGatewayClassConfigReconcile(t *testing.T) {
+	t.Parallel()
+	deletionTimestamp := meta.Now()
+	cases := []struct {
+		name       string
+		k8sObjects func() []runtime.Object
+		expErr     string
+		reque      bool
+		requeAfter time.Duration
+	}{
+		{
+			name: "Happy Path",
```

Comment:
Nit, could we call "Happy Path" "Successfully reconcile without any changes".

Reply:
Done